### ISSUE TITLE
Firebase Remote Config fetch() not called on fresh install in iOS

### ISSUE THREAD

**Issue by christianroman** [firebase/firebase-ios-sdk#11898]
### Introduction
I'm working on a high-profile iOS app where we've recently integrated Firebase Remote Config. We noticed that, even after correct SDK configuration, the `fetch()` method from Firebase Remote Config isn't invoked during a fresh launch. However, it works as expected on subsequent launches.

Here's my initialization code:
```swift
func application(_ application: UIApplication,
                 didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {

  FirebaseApp.configure()

  remoteConfig = RemoteConfig.remoteConfig()
  remoteConfig.setDefaults(fromPlist: "remote_config_defaults")

  let settings = RemoteConfigSettings()
  settings.minimumFetchInterval = 0
  settings.fetchTimeout = 10
  remoteConfig.configSettings = settings

  self.remoteConfig.fetch { status, error in
    print(error) // <- completion callback never gets called on a fresh install
    print(status)
  }
  ...
}
```

### Debugging Observations:
During debugging, I observed that the method `[analytics getUserPropertiesWithCallback:completionHandler]` from `RCNConfigFetch.m` isn't calling the `completionHandler`.

`RCNConfigFetch.m:`
```objective-c
- (void)getAnalyticsUserPropertiesWithCompletionHandler:
    (FIRAInteropUserPropertiesCallback)completionHandler {
  FIRLogDebug(kFIRLoggerRemoteConfig, @"I-RCN000060", @"Fetch with user properties completed.");
  id<FIRAnalyticsInterop> analytics = self->_analytics;
  if (analytics == nil) {
    completionHandler(@{});
  } else {
    [analytics getUserPropertiesWithCallback:completionHandler]; // <- completionHandler is never called.
  }
}
```

<img width="868" alt="Screenshot 2023-10-04 at 16 59 49" src="https://github.com/firebase/firebase-ios-sdk/assets/461021/756a4f1f-377d-4d9d-81d9-78fac716f5b2">

I attempted to use both SPM and manual xcframeworks imports, and even tested on different versions (10.15.0 and 10.13.0) of Firebase. The issue persists across all these setups.

### Environment / Other Frameworks Used

We use these frameworks:
- Analytics
- Crashlytics
- Performance

<img width="540" alt="Screenshot 2023-10-05 at 8 43 54" src="https://github.com/firebase/firebase-ios-sdk/assets/461021/def63b2d-595c-46fd-9f54-faf3a009f99a">

### Reproducing the issue

- Install the app afresh on an iOS device.
- Launch the app.
- Observe that the Firebase Remote Config fetch() completion callback is not invoked.

*Note:* I tried to replicate the same scenario on the sample project, but couldn't reproduce the issue there.

### Firebase SDK Version

10.15.0

### Xcode Version

14.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Remote Config

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json
{
  "identity" : "firebase-ios-sdk",
  "kind" : "remoteSourceControl",
  "location" : "https://github.com/firebase/firebase-ios-sdk.git",
  "state" : {
    "revision" : "8a8ec57a272e0d31480fb0893dda0cf4f769b57e",
    "version" : "10.15.0"
  }
}
```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
@christianroman Sorry about the issue and thanks for sharing the detailed analysis. It sounds like a duplicate of #11770 that was fixed in Firebase 10.16.0 that released earlier this week. Please upgrade and let us know if it addresses the issue.

**Comment by christianroman** [comment]
@paulb777 Thank you. We have just upgraded to version `10.16.0` and can confirm that the bug has been fixed. Thanks again!
